io-engine/src/grpc/v1/lvm/pool.rs

```diff
@@ -0,0 +1,178 @@
+use crate::{
+    grpc::{lvm_enabled, GrpcResult},
+    lvm::{CmnQueryArgs, Error as LvmError, VolumeGroup},
+    lvs::Lvs,
+    pool_backend::PoolArgs,
+};
+use io_engine_api::v1::pool::*;
+use std::{convert::TryFrom, fmt::Debug};
+use tonic::{Request, Response, Status};
+
+#[derive(Debug)]
+struct UnixStream(tokio::net::UnixStream);
+
+#[derive(Debug, Clone)]
+pub(crate) struct PoolService {}
+
+impl PoolService {
+    pub(crate) fn new() -> Self {
+        Self {}
+    }
+    pub(crate) async fn list_svc_pools(
+        &self,
+        args: &ListPoolOptions,
+    ) -> Result<Vec<Pool>, tonic::Status> {
+        lvm_enabled()?;
+
+        let pools = VolumeGroup::list(
+            &CmnQueryArgs::ours()
+                .named_opt(&args.name)
+                .uuid_opt(&args.uuid),
+        )
+        .await?;
+        Ok(pools.into_iter().map(Into::into).collect())
+    }
+}
+
+#[tonic::async_trait]
+impl PoolRpc for PoolService {
+    async fn create_pool(
+        &self,
+        request: Request<CreatePoolRequest>,
+    ) -> GrpcResult<Pool> {
+        lvm_enabled()?;
+
+        let args = PoolArgs::try_from(request.into_inner())?;
+
+        // bail if an lvs pool already exists with the same name
+        if let Some(_pool) = Lvs::lookup(args.name.as_str()) {
+            return Err(Status::invalid_argument(
+                "lvs pool with the same name already exists",
+            ));
+        };
+        // check if the disks are used by existing lvs pool
+        if Lvs::iter()
+            .map(|l| l.base_bdev().name().to_string())
+            .any(|d| args.disks.contains(&d))
+        {
+            return Err(Status::invalid_argument(
+                "an lvs pool already uses the disk",
+            ));
+        };
+        VolumeGroup::import_or_create(args)
```

Comment:
How the disk which is used for lvm pool will not be used in lvs pool?

Reply:
there's a simple check above, though I suspect that is not sufficient...
I think we need to try to import first as lvs and only if it fails try to import as lvm, this as another PR I think, this is already too large

Reply:
I mean do We need to have similar checks in `lvs::create_pool()` where if the lvm pool is already created with some disk, then same disk should not be used for `lvs pool` ?

Reply:
We'll use a common check for both cases, so it can be used on any pool_type